diffusion/learned.py

```diff
@@ -10,47 +10,104 @@
 class LearnedGaussianDiffusion(GaussianDiffusion):
     """Implements the core learning and inference algorithms."""
     def __init__(
-        self, noise_model, z_encoder_model, timesteps, img_shape,
```

Comment:
Okay, so it sounds like you're modifying this code to use a learned linear transformation, which is cool. However, are you not undoing the previous functionality of this class? (I feel like we are since we are getting rid of `z_encoder_model`, without which this class can't do what it has before. Can you maybe create a new type of `LearnedDiffusion` (a new class) that has the features that you're looking for, but would keep this old class (I see myself playing with it for example).